src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToAsyncMethodsTests.cs

```diff
@@ -0,0 +1,3272 @@
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_PassingExplicitCancellationTokenNone()
+        {
+            return VerifyCS.VerifyAnalyzerAsync(@"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await MethodAsync(CancellationToken.None);
+    }
+    Task MethodAsync() => Task.CompletedTask;
+    Task MethodAsync(CancellationToken c) => Task.CompletedTask;
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_OverloadTokenNotLastParameter()
+        {
+            return VerifyCS.VerifyAnalyzerAsync(@"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await MethodAsync();
+    }
+    Task MethodAsync() => Task.CompletedTask;
+    Task MethodAsync(int x, CancellationToken ct, string s) => Task.CompletedTask;
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_OverloadWithMultipleTokens()
+        {
+            return VerifyCS.VerifyAnalyzerAsync(@"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await MethodAsync();
+    }
+    Task MethodAsync() => Task.CompletedTask;
+    Task MethodAsync(CancellationToken c1, CancellationToken ct2) => Task.CompletedTask;
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_OverloadWithMultipleTokensSeparated()
+        {
+            return VerifyCS.VerifyAnalyzerAsync(@"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await MethodAsync();
+    }
+    Task MethodAsync() => Task.CompletedTask;
+    Task MethodAsync(int x, CancellationToken c1, string s, CancellationToken ct2) => Task.CompletedTask;
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_NamedTokenUnordered()
+        {
+            return VerifyCS.VerifyAnalyzerAsync(@"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await MethodAsync(s: ""Hello world"", c: CancellationToken.None, x: 5);
+    }
+    Task MethodAsync(int x, string s, CancellationToken c) => Task.CompletedTask;
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_Overload_NamedTokenUnordered()
+        {
+            return VerifyCS.VerifyAnalyzerAsync(@"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await MethodAsync(s: ""Hello world"", c: CancellationToken.None, x: 5);
+    }
+    Task MethodAsync(int x, string s) => Task.CompletedTask;
+    Task MethodAsync(int x, string s, CancellationToken c) => Task.CompletedTask;
+}
+            ");
+        }
+
+        #endregion
+
+        #region Diagnostics with no fix = C#
```

Comment:
@mavasani I found this case that I cannot get fixed, only diagnosed.
If the user calls a method with unordered named parameters, and the analyzer finds an overload that takes a ct, then in the fixer I have no way to add the argument name unless I look for the overload again, the find the ct parameter, and retrieve its name.
Is it ok to leave a case unfixed, or do all diagnosed cases need to be fixable?

Reply:
If I attempt to fix it by adding the argument without a name, then this shows up in the fixed code:
CA8323: Named argument 'z' is used out-of-position but is followed by an unnamed argument

Reply:
I wanted to add this unit test as a `VerifyCodeFixAsync` but it fails because the fixedCode gets analyzed as well. @mavasani explained to me via chat why it fails: I am registering a code fix but bailing in the middle of it. I need to modify my code so that I bail _before_ I register the code fix.

Reply:
Addressed. @mavasani can you please take a look? If I don't generate a new document root I now bail out early and don't register a code fix.
I am still unsure if the way I'm testing this in the unit tests is the best way. I tried again calling `VerifyCodeFixAsync` and the test is still analyzing the original code and the fixed code, and finding a diagnostic in the fixed code as expected, so I opted to use `VerifyAnalyzerAsync` instead.